src/wp-includes/media.php

```diff
@@ -2070,20 +2094,14 @@ function wp_img_tag_add_decoding_attr( $image, $context ) {
 		return $image;
 	}
 
-	/**
-	 * Filters the `decoding` attribute value to add to an image. Default `async`.
-	 *
-	 * Returning a falsey value will omit the attribute.
-	 *
-	 * @since 6.1.0
-	 *
-	 * @param string|false|null $value   The `decoding` attribute value. Returning a falsey value
-	 *                                   will result in the attribute being omitted for the image.
-	 *                                   Otherwise, it may be: 'async' (default), 'sync', or 'auto'.
-	 * @param string            $image   The HTML `img` tag to be filtered.
-	 * @param string            $context Additional context about how the function was called
-	 *                                   or where the img tag is.
+	/*
+	 * Only apply the decoding attribute to images that don't already have a decoding attribute.
 	 */
+	if ( str_contains( $image, ' decoding=' ) ) {
```

Comment:
This change shouldn't be made here, for two reasons:
* The deprecated function is no longer called, so it wouldn't make sense to update it as such. One could argue it's a breaking change, since before this function didn't bail if the attribute was already present.
* Deprecated functions generally should not be modified.

I left a related comment above for where this logic should be added instead, which is `wp_img_tag_add_loading_optimization_attrs()`, since that is the function that replaces this deprecated function and the call to it in `wp_filter_content_tags()`.